packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -173,19 +240,93 @@ class RawMenuAnchor extends StatefulWidget {
     this.consumeOutsideTaps = false,
     this.onOpen,
     this.onClose,
+    this.onOpenRequested = _defaultOnOpenRequested,
+    this.onCloseRequested = _defaultOnCloseRequested,
     this.useRootOverlay = false,
     this.builder,
     required this.controller,
     required this.overlayBuilder,
     this.child,
   });
 
-  /// A callback that is invoked when the menu is opened.
+  /// Called when the menu overlay is shown.
+  ///
+  /// This callback is triggered when the menu overlay is inserted into the
+  /// widget tree, typically before any opening animations begin. A typical
+  /// usage is to respond when the menu first becomes interactive, such as by
+  /// setting focus to a menu item.
+  ///
+  /// This callback is called regardless of whether the menu is already open or
+  /// the menu's position has changed.
   final VoidCallback? onOpen;
 
-  /// A callback that is invoked when the menu is closed.
+  /// Called when the menu overlay is hidden.
+  ///
+  /// This callback is triggered when the menu overlay is removed from the
+  /// widget tree, typically after any closing animations have completed.
+  ///
+  /// If a callback is provided for [onCloseRequested], [onClose] will not be
+  /// called until after `hideOverlay` is called.
   final VoidCallback? onClose;
 
+  /// Called when a request is made to open the menu.
+  ///
+  /// This callback is used by themed menu widgets to intercept open requests,
+  /// typically with the intent of adding a delay or an animation.
+  ///
```

Comment:
Somehow we've deviated from mentioning `MenuController.open`, which makes it feel confusing to me, since it is mentioned by many paragraphs below. Also combining another sentence from the end of this doc.
```suggestion
  ///
  /// This callback is triggered every time [MenuController.open] is called, including cases when
  /// the overlay is already showing, which might be useful if the menu is repositioned,
  /// or if the menu is opened during its closing animation.
  ///
```

Reply:
Awesome!